### PR TITLE
Updates to eaves measure.

### PR DESCRIPTION
**Pull request by joseph-robertson**

### CODE REVIEW

File: measures/ResidentialGeometryEaves/tests/create_residential_eaves_test.rb [hunk 1]
```diff
@@ -262,13 +261,17 @@ def _test_measure(osm_file_or_model, args_hash, expected_num_del_objects, expect
...
+    model.save(output_file_path, true)
+    
+    # show_output(result)
 
     # assert that it ran correctly
     assert_equal("Success", result.value.valueName)
-    assert_equal(num_infos, result.info.size)
-    assert_equal(num_warnings, result.warnings.size)
-    
+    # assert(result.info.size == num_infos)
```
Comment (shorowit) [inline measures/ResidentialGeometryEaves/tests/create_residential_eaves_test.rb]: Please uncomment.

File: measures/ResidentialGeometryEaves/tests/create_residential_eaves_test.rb [hunk 2]
```diff
@@ -287,18 +290,20 @@ def _test_measure(osm_file_or_model, args_hash, expected_num_del_objects, expect
...
+                w = OpenStudio.convert(w,"m","ft").get
                 if l < w
-                  next if UnitConversions.convert(l,"m","ft") > 5
-                  assert_in_epsilon(expected_values["eaves_depth"], UnitConversions.convert(l,"m","ft"), 0.01)
+                  next if OpenStudio::convert(l,"m","ft").get > 5
+                  assert_in_epsilon(expected_values["eaves_depth"], l, 0.01)
                 else
-                  next if UnitConversions.convert(w,"m","ft") > 5
-                  assert_in_epsilon(expected_values["eaves_depth"], UnitConversions.convert(w,"m","ft"), 0.01)
+                  next if OpenStudio::convert(w,"m","ft").get > 5
```
Comment (shorowit) [inline measures/ResidentialGeometryEaves/tests/create_residential_eaves_test.rb]: Please use UnitConversions in the four lines above.

File: measures/ResidentialGeometryEaves/tests/create_residential_eaves_test.rb [hunk 3]
```diff
@@ -262,13 +261,17 @@ def _test_measure(osm_file_or_model, args_hash, expected_num_del_objects, expect
     measure.run(model, runner, argument_map)
     result = runner.result
     
-    #show_output(result)
+    # save the model to test output directory
+    output_file_path = OpenStudio::Path.new(File.dirname(__FILE__) + "/output/#{test_name}.osm")
+    model.save(output_file_path, true)
```
Comment (shorowit) [inline measures/ResidentialGeometryEaves/tests/create_residential_eaves_test.rb]: Remove this debug code?

File: measures/ResidentialGeometryEaves/measure.rb [hunk 4]
```diff
@@ -2,7 +2,6 @@
 # http://nrel.github.io/OpenStudio-user-documentation/measures/measure_writing_guide/
 
 require "#{File.dirname(__FILE__)}/resources/geometry"
-require "#{File.dirname(__FILE__)}/resources/unit_conversions"
```
Comment (shorowit) [inline measures/ResidentialGeometryEaves/measure.rb]: Add back in.

File: measures/ResidentialGeometryEaves/measure.rb [hunk 5]
```diff
@@ -59,7 +58,7 @@ def run(model, runner, user_arguments)
     end
 
     roof_structure = runner.getStringArgumentValue("roof_structure",user_arguments)
-    eaves_depth = UnitConversions.convert(runner.getDoubleArgumentValue("eaves_depth",user_arguments),"ft","m")
+    eaves_depth = OpenStudio.convert(runner.getDoubleArgumentValue("eaves_depth",user_arguments),"ft","m").get
```
Comment (shorowit) [inline measures/ResidentialGeometryEaves/measure.rb]: Use UnitConversions

File: measures/ResidentialGeometryEaves/measure.rb [hunk 6]
```diff
@@ -72,7 +71,6 @@ def run(model, runner, user_arguments)
         existing_eaves_depth = get_existing_eaves_depth(shading_surface)
       end
       shading_surface_group.remove
-      runner.registerInfo("Removed existing #{Constants.ObjectNameEaves}.")
```
Comment (shorowit) [inline measures/ResidentialGeometryEaves/measure.rb]: Remove debug statement.

File: measures/ResidentialGeometryEaves/measure.rb [hunk 7]
```diff
@@ -85,1474 +83,254 @@ def run(model, runner, user_arguments)
...
-              new_vertices_left = Geometry.make_polygon(sw_point, nw_point, ne_point, se_point)    
-
-              sw_point = OpenStudio::Point3d.new(bottom_right.x,bottom_right.y,bottom_right.z+z_offset)
-              nw_point = OpenStudio::Point3d.new(top_right.x,top_right.y,top_right.z+z_offset)
-              ne_point = OpenStudio::Point3d.new(top_right.x+eaves_depth,top_right.y,top_right.z+z_offset)
-              se_point = OpenStudio::Point3d.new(bottom_right.x+eaves_depth,bottom_right.y,bottom_right.z+z_offset)
-              new_vertices_right = Geometry.make_polygon(sw_point, nw_point, ne_point, se_point)    
-
-              sw_point = OpenStudio::Point3d.new(bottom_left.x-eaves_depth,bottom_left.y-eaves_depth,bottom_left.z+z_offset)
-
```
Comment (shorowit) [inline measures/ResidentialGeometryEaves/measure.rb]: What about multifamily?

Comment (joseph-robertson) [reply measures/ResidentialGeometryEaves/measure.rb]: Not sure what you mean. ?

File: measures/ResidentialGeometryEaves/measure.rb (hunk 6 above)
Comment (joseph-robertson) [reply measures/ResidentialGeometryEaves/measure.rb]: My version didn't have this line. Had that already been removed?

File: measures/ResidentialGeometryEaves/tests/create_residential_eaves_test.rb (hunk 3 above)
Comment (joseph-robertson) [reply measures/ResidentialGeometryEaves/tests/create_residential_eaves_test.rb]: Commented out.

File: measures/ResidentialGeometryEaves/tests/create_residential_eaves_test.rb (hunk 1 above)
Comment (joseph-robertson) [reply measures/ResidentialGeometryEaves/tests/create_residential_eaves_test.rb]: Oops.

File: measures/ResidentialGeometryEaves/measure.rb (hunk 6 above)
Comment (shorowit) [reply measures/ResidentialGeometryEaves/measure.rb]: Oops, this was a line you removed, not added. My bad :-)

File: measures/ResidentialGeometryEaves/measure.rb (hunk 7 above)
Comment (shorowit) [reply measures/ResidentialGeometryEaves/measure.rb]: Hmm, somehow this got associated with the wrong line. You have a line in the code that does something specifically for SFA homes, I wondered whether it should also apply to MF.

Comment (joseph-robertson) [reply measures/ResidentialGeometryEaves/measure.rb]: This measure should accommodate SFA and MF. And there are a few unit tests to check that it does.

**Pull request merged**